### DISCUSSION TITLE
How to get product by handle

### DISCUSSION THREAD

**Discussion by winston0410** [medusajs/medusa#4325] [Q&A] [answered]
From the current API, only `id` is accepted for searching product: https://docs.medusajs.com/api/store#tag/Products/operation/GetProductsProduct

I want to search a product by its handle, so I can show the user a url of `/product/product-handle-foo-bar`, instead of `/product/prod_01H30AHABK5V91AP7BPTJ1JJCC`. Is this feature missing or possible with the current API?

**Comment by tsvetann** [comment, marked as answer]
`await medusaClient.products.list({handle})`?

**Reply by tsvetann** [reply]
AFAIK there is no product retrieve function by handle or at least I am not aware of one

**Reply by winston0410** [reply]
Indeed that would work for now. Thanks!

**Reply by henzyd** [reply]
How would I do that using curl?
